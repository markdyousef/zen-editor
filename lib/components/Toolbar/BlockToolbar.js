'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _draftJs = require('draft-js');

var _StyleButton = require('../StyleButton');

var _StyleButton2 = _interopRequireDefault(_StyleButton);

var _h = require('../../icons/h1');

var _h2 = _interopRequireDefault(_h);

var _h3 = require('../../icons/h2');

var _h4 = _interopRequireDefault(_h3);

var _quote = require('../../icons/quote');

var _quote2 = _interopRequireDefault(_quote);

var _unorderedList = require('../../icons/unorderedList');

var _unorderedList2 = _interopRequireDefault(_unorderedList);

var _codeBlock = require('../../icons/codeBlock');

var _codeBlock2 = _interopRequireDefault(_codeBlock);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BLOCK_TYPES = [{ label: 'H1', style: 'header-one', Icon: _h2.default }, { label: 'H2', style: 'header-two', Icon: _h4.default }, { label: 'Quote', style: 'blockquote', Icon: _quote2.default }, { label: 'Bulleted List', style: 'unordered-list-item', Icon: _unorderedList2.default }, { label: 'Numbered List', style: 'ordered-list-item', Icon: _unorderedList2.default }, { label: 'Code Block', style: 'code-block', Icon: _codeBlock2.default }];

var BlockToolbar = function BlockToolbar(_ref) {
    var onToggle = _ref.onToggle,
        editorState = _ref.editorState;

    var blockType = _draftJs.RichUtils.getCurrentBlockType(editorState);
    return _react2.default.createElement(
        'div',
        null,
        BLOCK_TYPES.map(function (item) {
            return _react2.default.createElement(_StyleButton2.default, {
                key: item.label,
                active: item.style === blockType,
                label: item.label,
                onToggle: onToggle,
                style: item.style,
                Icon: item.Icon
            });
        })
    );
};

BlockToolbar.propTypes = {
    editorState: _react.PropTypes.shape({
        _immutable: _react.PropTypes.object
    }).isRequired,
    onToggle: _react.PropTypes.func.isRequired
};

exports.default = BlockToolbar;