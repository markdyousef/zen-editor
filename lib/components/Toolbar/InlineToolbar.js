'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _StyleButton = require('../StyleButton');

var _StyleButton2 = _interopRequireDefault(_StyleButton);

var _bold = require('../../icons/bold');

var _bold2 = _interopRequireDefault(_bold);

var _italic = require('../../icons/italic');

var _italic2 = _interopRequireDefault(_italic);

var _underline = require('../../icons/underline');

var _underline2 = _interopRequireDefault(_underline);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var INLINE_BUTTONS = [{
    label: 'B',
    style: 'BOLD',
    Icon: _bold2.default,
    description: 'Bold'
}, {
    label: 'I',
    style: 'ITALIC',
    Icon: _italic2.default,
    description: 'Italic'
}, {
    label: 'U',
    style: 'UNDERLINE',
    Icon: _underline2.default,
    description: 'Underline'
}, {
    label: 'Hi',
    style: 'HIGHLIGHT',
    Icon: _underline2.default,
    description: 'Highlight selection'
}];


var InlineToolbar = function InlineToolbar(_ref) {
    var onToggle = _ref.onToggle,
        editorState = _ref.editorState;

    var currentStyle = editorState.getCurrentInlineStyle();
    return _react2.default.createElement(
        'div',
        null,
        INLINE_BUTTONS.map(function (item) {
            return _react2.default.createElement(_StyleButton2.default, {
                key: item.label,
                active: currentStyle.has(item.style),
                label: item.label,
                onToggle: onToggle,
                style: item.style,
                Icon: item.Icon
            });
        })
    );
};

InlineToolbar.propTypes = {
    editorState: _react.PropTypes.shape({
        _immutable: _react.PropTypes.object
    }).isRequired,
    onToggle: _react.PropTypes.func.isRequired
};

exports.default = InlineToolbar;