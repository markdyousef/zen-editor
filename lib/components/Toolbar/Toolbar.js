'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.inlineToolbarPlugin = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _draftJsInlineToolbarPlugin = require('draft-js-inline-toolbar-plugin');

var _draftJsInlineToolbarPlugin2 = _interopRequireDefault(_draftJsInlineToolbarPlugin);

var _draftJsButtons = require('draft-js-buttons');

var _buttonsStyles = require('./styles/buttonsStyles.css');

var _buttonsStyles2 = _interopRequireDefault(_buttonsStyles);

var _toolbarStyles = require('./styles/toolbarStyles.css');

var _toolbarStyles2 = _interopRequireDefault(_toolbarStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import 'draft-js-inline-toolbar-plugin/lib/plugin.css';
var inlineToolbarPlugin = exports.inlineToolbarPlugin = (0, _draftJsInlineToolbarPlugin2.default)({
    structure: [_draftJsButtons.BoldButton, _draftJsButtons.ItalicButton, _draftJsButtons.UnderlineButton, _draftJsButtons.HeadlineOneButton, _draftJsButtons.HeadlineTwoButton, _draftJsButtons.UnorderedListButton, _draftJsButtons.OrderedListButton, _draftJsButtons.BlockquoteButton, _draftJsButtons.CodeBlockButton],
    theme: { buttonStyles: _buttonsStyles2.default, toolbarStyles: _toolbarStyles2.default }
});

var InlineToolbar = inlineToolbarPlugin.InlineToolbar;

exports.default = function () {
    return _react2.default.createElement(InlineToolbar, null);
};