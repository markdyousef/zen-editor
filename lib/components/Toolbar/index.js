'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.inlineToolbarPlugin = undefined;

var _Toolbar = require('./Toolbar');

var _Toolbar2 = _interopRequireDefault(_Toolbar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.inlineToolbarPlugin = _Toolbar.inlineToolbarPlugin;
exports.default = _Toolbar2.default;