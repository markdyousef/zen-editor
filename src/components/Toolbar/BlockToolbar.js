// @flow
import React, { PropTypes } from 'react';
import { RichUtils } from 'draft-js';
import StyleButton from './StyleButton';

const BLOCK_TYPES = [
  { label: 'H1', style: 'header-one' },
  { label: 'H2', style: 'header-two' },
  { label: 'H3', style: 'header-three' },
  { label: 'Quote', style: 'blockquote' },
  { label: 'Bulleted List', style: 'unordered-list-item' },
  { label: 'Numbered List', style: 'ordered-list-item' },
  { label: 'Code Block', style: 'code-block' }
];

const BlockToolbar = ({ onToggle, editorState }:Object) => {
    const blockType = RichUtils.getCurrentBlockType(editorState);
    return (
        <div>
            {BLOCK_TYPES.map(block =>
                <StyleButton
                    key={block.label}
                    active={block.style === blockType}
                    label={block.label}
                    onToggle={onToggle}
                    style={block.style}
                />
            )}
        </div>
    );
};

BlockToolbar.propTypes = {
    editorState: PropTypes.shape({
        _immutable: PropTypes.object
    }).isRequired,
    onToggle: PropTypes.func.isRequired
};

export default BlockToolbar;
