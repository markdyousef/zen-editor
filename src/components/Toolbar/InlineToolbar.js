// @flow
import React, { PropTypes } from 'react';
import StyledButton from './StyleButton';

const INLINE_BUTTONS = [
    {
        label: 'B',
        style: 'BOLD',
        icon: 'bold',
        description: 'Bold'
    },
    {
        label: 'I',
        style: 'ITALIC',
        icon: 'italic',
        description: 'Italic'
    },
    {
        label: 'U',
        style: 'UNDERLINE',
        icon: 'underline',
        description: 'Underline'
    },
    {
        label: 'Hi',
        style: 'HIGHLIGHT',
        description: 'Highlight selection'
    }
];

const InlineToolbar = ({ onToggle, editorState }:Object) => {
    const currentStyle = editorState.getCurrentInlineStyle();
    return (
        <div>
            {INLINE_BUTTONS.map(item =>
                <StyledButton
                    key={item.label}
                    active={currentStyle.has(item.style)}
                    label={item.label}
                    onToggle={onToggle}
                    style={item.style}
                />
            )}
        </div>
    );
};

InlineToolbar.propTypes = {
    editorState: PropTypes.shape({
        _immutable: PropTypes.object
    }).isRequired,
    onToggle: PropTypes.func.isRequired
};

export default InlineToolbar;
