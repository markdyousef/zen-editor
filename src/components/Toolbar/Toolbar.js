import React from 'react';
// import 'draft-js-inline-toolbar-plugin/lib/plugin.css';
import createInlineToolbarPlugin from 'draft-js-inline-toolbar-plugin';
import {
  ItalicButton,
  BoldButton,
  UnderlineButton,
  HeadlineOneButton,
  HeadlineTwoButton,
  UnorderedListButton,
  OrderedListButton,
  BlockquoteButton,
  CodeBlockButton
} from 'draft-js-buttons';
import buttonStyles from './styles/buttonsStyles.css';
import toolbarStyles from './styles/toolbarStyles.css';

export const inlineToolbarPlugin = createInlineToolbarPlugin({
    structure: [
        BoldButton,
        ItalicButton,
        UnderlineButton,
        HeadlineOneButton,
        HeadlineTwoButton,
        UnorderedListButton,
        OrderedListButton,
        BlockquoteButton,
        CodeBlockButton
    ],
    theme: { buttonStyles, toolbarStyles }
});

const { InlineToolbar } = inlineToolbarPlugin;

export default () => {
    return <InlineToolbar />;
};
