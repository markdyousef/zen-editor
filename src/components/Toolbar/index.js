import Toolbar, { inlineToolbarPlugin } from './Toolbar';

export {
    inlineToolbarPlugin
};

export default Toolbar;
