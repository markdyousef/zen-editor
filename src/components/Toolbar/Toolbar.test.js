import React from 'react';
import renderer from 'react-test-renderer';
import Toolbar from './Toolbar';

it('renders without crashing', () => {
    const props = {
        onToggle: () => {},
        editorState: {}
    };
    const component = renderer.create(<Toolbar {...props} />);
    expect(component).toMatchSnapshot(component);
});
