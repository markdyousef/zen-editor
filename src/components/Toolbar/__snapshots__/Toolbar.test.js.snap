exports[`[object Object] 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Decorated(Toolbar) />,
  "nodes": Array [
    <Decorated(Toolbar) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        editorState={Object {}}
        focus={[Function]}
        showToolbar={true}
        toggleBlockType={[Function]}
        toggleInlineStyle={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "editorState": Object {},
          "focus": [Function],
          "showToolbar": true,
          "toggleBlockType": [Function],
          "toggleInlineStyle": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Decorated(Toolbar) />,
        "_debugID": 2,
        "_renderedOutput": <Decorated(Toolbar) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    editorState={Object {}}
    focus={[Function]}
    showToolbar={true}
    toggleBlockType={[Function]}
    toggleInlineStyle={[Function]} />,
}
`;
